iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -46,16 +43,106 @@ protected String getTypeDescription() {
         return "Shunt compensator";
     }
 
+    class ShuntCompensatorLinearModelAdderImpl implements ShuntCompensatorLinearModelAdder {
+
+        private double bPerSection = Double.NaN;
+
+        private double gPerSection = Double.NaN;
+
+        private int maximumSectionCount = -1;
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setbPerSection(double bPerSection) {
+            this.bPerSection = bPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setgPerSection(double gPerSection) {
+            this.gPerSection = gPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setMaximumSectionCount(int maximumSectionCount) {
+            this.maximumSectionCount = maximumSectionCount;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorAdder add() {
+            ValidationUtil.checkbPerSection(ShuntCompensatorAdderImpl.this, bPerSection);
+            ValidationUtil.checkMaximumSectionCount(ShuntCompensatorAdderImpl.this, maximumSectionCount);
+            model = new ShuntCompensatorLinearModelImpl(bPerSection, gPerSection, maximumSectionCount);
+            return ShuntCompensatorAdderImpl.this;
+        }
+    }
+
+    class ShuntCompensatorNonLinearModelAdderImpl implements ShuntCompensatorNonLinearModelAdder {
+
+        private final TreeMap<Integer, ShuntCompensatorNonLinearModelImpl.SectionImpl> sections = new TreeMap<>();
+
+        class SectionAdderImpl implements SectionAdder {
+
+            private int sectionNum = -1;
+
+            private double b = Double.NaN;
+
+            private double g = Double.NaN;
+
+            @Override
+            public SectionAdder setSectionNum(int sectionNum) {
+                this.sectionNum = sectionNum;
+                return this;
+            }
+
+            @Override
+            public SectionAdder setB(double b) {
+                this.b = b;
+                return this;
+            }
+
+            @Override
+            public SectionAdder setG(double g) {
+                this.g = g;
+                return this;
+            }
+
+            @Override
+            public ShuntCompensatorNonLinearModelAdder endSection() {
+                ValidationUtil.checkSectionNumber(ShuntCompensatorAdderImpl.this, sectionNum);
+                if (sections.containsKey(sectionNum)) {
+                    throw new ValidationException(ShuntCompensatorAdderImpl.this, "a section is already defined at this number");
+                }
```

Comment:
It's strange to not replace the section whereas it's totally allowed later? It's not really important but it's questioning me

Reply:
Not sure if choosing to replace a section in a built model and creating the same section twice while building the whole model is the same thing? I think the second case is more probably an error but I may be mistaken. We can delete this validation condition and consider it is allowed to rewrite a section while building the model.